java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -1027,22 +1027,36 @@ private static JavaType getPrimitive(JavaType primitiveOrWrapper) {
   }
 
   public Optional<JavaSymbol.MethodJavaSymbol> getSamMethod(JavaType lambdaType) {
-    for (Symbol member : lambdaType.symbol().memberSymbols()) {
-      if (isAbstractMethod(member)) {
-        JavaSymbol.MethodJavaSymbol methodJavaSymbol = (JavaSymbol.MethodJavaSymbol) member;
-        boolean isObjectMethod = isObjectMethod(methodJavaSymbol);
-        if(!isObjectMethod) {
-          return Optional.of(methodJavaSymbol);
-        }
-      }
+    List<JavaSymbol.MethodJavaSymbol> methodSymbols = abstractMethodsOfType(lambdaType);
+    if (methodSymbols.size() > 1) {
+      return Optional.empty();
     }
-    for (ClassJavaType type : lambdaType.symbol.superTypes()) {
-      Optional<JavaSymbol.MethodJavaSymbol> samMethod = getSamMethod(type);
-      if (samMethod.isPresent()) {
-        return samMethod;
+    if (methodSymbols.size() == 1) {
+      JavaSymbol.MethodJavaSymbol override = methodSymbols.get(0).overriddenSymbol();
+      Optional<JavaSymbol.MethodJavaSymbol> otherAbstractMethod = lambdaType.symbol.superTypes()
+        .stream()
+        .filter(t -> t.symbol().isInterface())
+        .flatMap(t -> abstractMethodsOfType(t).stream())
+        .filter(m -> override == null || !override.equals(m))
```

Comment:
We could add cases to the `Lambdas.java` test to cover these cases already, it seems it is not entirely covered by current tests